lib/sudo-cutils/src/lib.rs

```diff
@@ -53,6 +56,20 @@ pub unsafe fn string_from_ptr(ptr: *const libc::c_char) -> String {
     }
 }
 
+/// Create an `OsString` copy from a C string pointer.
+///
+/// # Safety
+/// This function assumes that the pointer is either a null pointer or that
+/// it points to a valid NUL-terminated C string.
+pub unsafe fn os_string_from_ptr(ptr: *const libc::c_char) -> OsString {
```

Comment:
Going to duplicate the unit test for string_from_ptr here; unsafe code should be miri-tested.